tests/test.cpp

```diff
@@ -3416,6 +3416,78 @@ void TestEmbeddedBinarySchema() {
           0);
 }
 
+void NullableScalarsTest() {
+  // Simple schemas and a "has nullable scalar" sentinal.
+  std::vector<std::string> schemas;
+  schemas.push_back("table Monster { mana : int; }");
+  schemas.push_back("table Monster { mana : int = 42; }");
+  schemas.push_back("table Monster { mana : int =  null; }");
+  schemas.push_back("table Monster { mana : long; }");
+  schemas.push_back("table Monster { mana : long = 42; }");
+  schemas.push_back("table Monster { mana : long = null; }");
+  schemas.push_back("table Monster { mana : float; }");
+  schemas.push_back("table Monster { mana : float = 42; }");
+  schemas.push_back("table Monster { mana : float = null; }");
+  schemas.push_back("table Monster { mana : double; }");
+  schemas.push_back("table Monster { mana : double = 42; }");
+  schemas.push_back("table Monster { mana : double = null; }");
+  schemas.push_back("table Monster { mana : bool; }");
+  schemas.push_back("table Monster { mana : bool = 42; }");
+  schemas.push_back("table Monster { mana : bool = null; }");
+
+  // Check the FieldDef is correctly set.
+  for (auto schema = schemas.begin(); schema < schemas.end(); schema++) {
+    const bool has_null = schema->find("null") != std::string::npos;
+    flatbuffers::Parser parser;
+    TEST_ASSERT(parser.Parse(schema->c_str()));
+    const auto *mana = parser.structs_.Lookup("Monster")->fields.Lookup("mana");
+    TEST_EQ(mana->nullable, has_null);
+  }
+
+  // Test if nullable scalars are allowed for each language.
+  const int kNumLanguages = 17;
+  for (int lang=0; lang<kNumLanguages; lang++) {
+    flatbuffers::IDLOptions opts;
+    opts.lang_to_generate |= 1 << lang;
+    for (auto schema = schemas.begin(); schema < schemas.end(); schema++) {
+      const bool has_null = schema->find("null") != std::string::npos;
+      flatbuffers::Parser parser(opts);
+      // Its not supported in any language yet so has_null means error.
+      TEST_EQ(parser.Parse(schema->c_str()), !has_null);
+    }
+  }
+}
+
+void JsonNullTest() {
+  {
+    char json[] = "{\"opt_field\": 123 }";
+    flatbuffers::Parser parser;
+    flexbuffers::Builder flexbuild;
+    parser.ParseFlexBuffer(json, nullptr, &flexbuild);
+    auto root = flexbuffers::GetRoot(flexbuild.GetBuffer());
```

Comment:
I guess the reason to use FlexBuffers is that it is easier than FlatBuffers and exercises the same function? If so, maybe a comment to that effect, since `JsonNullTest` is far too generic for what happens here.

Reply:
Are you suggesting there is a better way to test these code paths through the public API? If so, how do you suggest?

Reply:
You could parse JSON with FlatBuffers instead? Not saying that is necessary, but at least a comment here what this test is for would be good.. the current function name is confusing I think.

Reply:
I suppose it would not work with `flatbuffres::Parser`.
This parser uses a schema while parsing JSON. The `null` isn't a valid integer or floating-point scalar.
A complete test should parse a JSON with flatbuffers::Parser and `GenerateText()` form it.

Reply:
Yea that's the bit I'm confused about -- I'm not sure where to inject behavior to get around `SupportsNullableScalars` and get a high fidelity test while this isn't implemented in any language. I'll leave a TODO for a future test instead.